### DISCUSSION TITLE
'Thin' Porter Bundle

### DISCUSSION THREAD

**Discussion by samjohnsonCF** [getporter/porter#2707] [Q&A] [answered]
Is there any way to make a 'thin' porter bundle? 
For example,
I have a lot of images that are referenced and pulled into the bundle when publishing a bundle. Is there anyway to not pull those images in(only reference them), but to just let k8s pull at runtime instead? This would allow for much faster build times and testing of bundles and changes.

**Comment by carolynvs** [comment, marked as answer]
Hi @samjohnsonCF, good question! When you include references to an image in your porter.yaml file using the [image](https://getporter.org/bundle/manifest/#images) section, that instructs Porter to make a thick bundle, including those images in the final bundle that is published to a registry or archived to a file. When you omit the images section, you create a thin bundle, where the images used by your bundle are not tracked and moved together with the bundle.

So if you do not need Porter to help relocate the images used in your manifests, you can safely omit them and speed up the build and publish steps.

**Reply by samjohnsonCF** [reply]
Okay cool! That is what I was figuring. 
If I wanted to still pass in the repo/tags of those images, I assume using the custom section would be the best solution to that scenario?

**Reply by carolynvs** [reply]
The custom section is locked-in a build time. So if you know the location of the images when you run `porter build` you can use the custom section to pass that data into the bundle, either by editing the porter.yaml or using the `--custom` flag defined on porter build.

If you want to have the image location be modifiable when installing the bundle, go with a parameter instead. Then you can specify the image locations like so: `porter install --param myimageref=docker.io/mystuff/myimage:v1.2.3`

**Reply by samjohnsonCF** [reply]
Ahh, a param would make more sense! Thank you for the info!
